src/Database/Query.php

```diff
@@ -1746,6 +1747,16 @@ public function newExpr($rawExpression = null)
         return $expression;
     }
 
+    /**
+     * Returns a new CaseExpression object to build SQL Case clauses
+     *
+     * @return \Cake\Database\Expression\CaseExpression
+     */
+    public function newCaseExpr()
```

Comment:
Why is this method here and not on QueryExpression like the other expression builders?

Reply:
Based on my [original comment](https://github.com/cakephp/cakephp/issues/11793#issuecomment-370888809), I try to make that code working.
`$query->newExpr()->newCaseExpr()` would be more verbose I think